### PR TITLE
[RISCV] Tuple intrinsics are creating overly aligned memory operands

### PR DESCRIPTION
**Pull request by 4vtomat**
The alignment should be same as its element type.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -1623,10 +1623,17 @@ bool RISCVTargetLowering::getTgtMemIntrinsic(IntrinsicInfo &Info,
       MemTy = MemTy->getScalarType();
 
     Info.memVT = getValueType(DL, MemTy);
-    if (MemTy->isTargetExtTy())
+    if (MemTy->isTargetExtTy()) {
+      // RISC-V vector tuple type's alignment type should be its element type.
+      if (cast<TargetExtType>(MemTy)->getName() == "riscv.vector.tuple")
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:1628]: (We definitely should fix this `strcmp`)

**Review by wangpc-pp — APPROVED** [review]
LGTM overall.

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -1623,10 +1623,17 @@ bool RISCVTargetLowering::getTgtMemIntrinsic(IntrinsicInfo &Info,
...
+        MemTy = Type::getIntNTy(
+            MemTy->getContext(),
+            1 << cast<ConstantInt>(I.getArgOperand(I.arg_size() - 1))
+                     ->getZExtValue());
       Info.align = DL.getABITypeAlign(MemTy);
-    else
+    } else {
       Info.align = Align(DL.getTypeSizeInBits(MemTy->getScalarType()) / 8);
+    }
```
[the remark this answers is not in the document]
Comment (4vtomat) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:1636]: I think we need to add to else block to keep it uniform with the `if` block: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp (hunk 1 above)
Comment (4vtomat) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:1628]: Yeah, maybe we should use enum instead of string lol

**Pull request merged**